### PR TITLE
add support for ALTER TABLE ... RENAME COLUMN syntax

### PR DESCRIPTION
**Pull request by tangwz**
#### Fix ALTER TABLE ... RENAME COLUMN syntax

Issue: pingcap/parser#462

#### MySQL Syntax:

```
alter_specification:
...
  | RENAME COLUMN old_col_name TO new_col_name
...
```

### Bad SQL Case:

```
ALTER TABLE t RENAME COLUMN a TO b;
```

### Check List

Tests
- Unit test

### CODE REVIEW

File: parser.y [hunk 1]
```diff
@@ -1526,6 +1526,14 @@ AlterTableSpec:
 			NewColumns:	[]*ast.ColumnDef{colDef},
 		}
 	}
+|   "RENAME" "COLUMN" ColumnName "TO" ColumnName
```
Comment (leoppro) [inline parser.y]: use `tab` but not `space` in L1529-1536

Comment (leoppro) [reply parser.y]: rest LGTM

Comment (tangwz) [reply parser.y]: I will fix it.

**Comment by tangwz** [comment]
@leoppro PATL

**Pull request merged**